content/en/agent/troubleshooting/hostname_containers.md

```diff
@@ -0,0 +1,240 @@
+        - name: agent
+          env:
+            - name: DD_KUBELET_TLS_VERIFY
+              value: "false"
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Accessing the cloud provider metadata endpoint
+
+If you run in AWS, GCP, or Azure, the Agent can use a metadata endpoint to retrieve the hostname.
+
+Accessing the cloud provider metadata endpoint properly matches Agent data and cloud integration data.
+
+Encountering this issue usually means that access to the metadata endpoint has been restricted.
+For example, on AWS, this happens through the [hop limit feature][5].
+
+### Accessing the container runtime API
+
+Use this solution only in the unlikely even that you **explicitly** don't want the Agent to connect to Kubelet API, and if you are not running in a supported cloud provider described above.
+
+In this case you can use the downward API to set `DD_HOSTNAME`:
+
+{{< tabs >}}
+{{% tab "Helm" %}}
+
+Custom `values.yaml`:
+
+```yaml
+datadog:
+  env:
+    - name: DD_HOSTNAME
+      valueFrom:
+        fieldRef:
+          fieldPath: spec.nodeName
+```
+
+{{% /tab %}}
+{{% tab "Operator" %}}
+
+`DatadogAgent` Kubernetes Resource:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog
+spec:
+  agent:
+    env:
+      - name: DD_HOSTNAME
+        valueFrom:
+          fieldRef:
+            fieldPath: spec.nodeName
+```
+
+{{% /tab %}}
+{{% tab "Manifest" %}}
+
+`DaemonSet` manifest
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: datadog
+spec:
+  template:
+    spec:
+      containers:
+        - name: agent
+          env:
+            - name: DD_HOSTNAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## AWS ECS and Docker VM hostname errors
+
+When the Agent runs in Docker on a cloud provider, a hostname error usually means that the Agent cannot access at least one of:
+* Container runtime API
+* Cloud provider metadata endpoint
+
+### Accessing the container runtime API
+
+Allow the Agent to connect to the Docker socket:
+
+{{< tabs >}}
+{{% tab "AWS ECS on EC2" %}}
+
+Make sure the Docker socket is mounted in your [task definition][1].
+
+
+[1]: /resources/json/datadog-agent-ecs.json
+{{% /tab %}}
+{{% tab "Docker on VM" %}}
+
+Make sure the Docker socket is mounted in your `docker run` command:
+
+```
+-v /var/run/docker.sock:/var/run/docker.sock:ro
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Accessing the cloud provider metadata endpoint
+
+If you're running in AWS, GCP, or Azure, the Agent can use metadata endpoint to retrieve the hostname.
+
+Accessing the cloud provider metadata endpoint properly matches Agent data and cloud integration data.
+
+Encountering this issue usually means that access to the metadata endpoint has been restricted.
+For example, on AWS, this happens through the [hop limit feature][5].
+
+## Hostname errors in CI environments and sidecar setups
```

Comment:
@vboulineau 
I suppose for EKS-Fargate because the agent have access to the kubelet api, It is not requires to set DD_HOSTAME. (also because it is fargate and in that case we want `host:""`)
maybe a small "note" in this section for the EKS-fargate case.

Reply:
I've added a small note on serverless solutions like Fargate.